File: java/slice-by-layer/gradle/core/src/main/java/com/example/catalog/core/ShowCatalog.java
package com.example.catalog.core;

import java.util.List;

public class ShowCatalog {
    private final CatalogRepo repo;
    private final CatalogItemPresenter presenter;

    public ShowCatalog(CatalogRepo repo, CatalogItemPresenter presenter) {
        this.repo = repo;
        this.presenter = presenter;
    }

    public void execute() {
        List<CatalogItem> catalogItems = repo.catalogItems();

        for (CatalogItem item: catalogItems) {
            presenter.presentCatalogItem(item);
        }
    }
}

File: go/gomod/core/show_catalog.go
package core

type showCatalog struct {
	repo          CatalogRepo
	itemPresenter CatalogItemPresenter
}

func ShowCatalog(repo CatalogRepo, presenter CatalogItemPresenter) *showCatalog {
	action := new(showCatalog)
	action.repo = repo
	action.itemPresenter = presenter
	return action
}

func (catalog *showCatalog) Execute() {
	catalogItems := catalog.repo.CatalogItems()
	for _, item := range catalogItems {
		catalog.itemPresenter.PresentCatalogItem(item)
	}
}